### ISSUE TITLE
javac ignores org.eclipse.jdt.annotation.Nullable annotation

### ISSUE THREAD

**Issue by marc-guenther** [immutables/immutables#1288]
We use the `@Nullable` annotation from `org.eclipse.jdt.annotation` like so:
```
abstract public @org.eclipse.jdt.annotation.Nullable String name();
```

or even
```
@Value.Default
@org.eclipse.jdt.annotation.Nullable
public String name() {
    return null;
}
```

We had some problem with this (which now I cannot reproduce), where this worked fine during _Eclipse_ incremental compilation, but when we triggered a Clean Build, the generated source code would ignore the annotation and include an `Objects.nonNull()` check. Doing any tiny change to the java file caused the null check to disappear.

This was quite annoying, but we quickly found a workaround, we also added the `@javax.annotation.Nullable`, and the problem went away, and so I never got around to report this.

But now we have a similar problem **without any workaround**. This time we have the annotation on the generic type of a _List_:

```
abstract public List<@org.eclipse.jdt.annotation.Nullable String> emails();
```
So we have a List, which can contain `null` values. _Immutables_ immediately warned us, that Guava Collections do not support `null`, so we switched on `jdkOnly = true`.

This works perfectly fine in _Eclipse_, but when doing a CI build using _javac_, it **ignores the annotation**, and generates a `null`-check, causing everything to fail.

Unfortunately, we cannot use the `javax.annotation.Nullable` here, as for whatever reason, it isn't allowed in this location.

I compared the two generated files, and they look quite different. The most notable difference is of course the null check.

This is more or less a blocker for us. I guess we could try to find yet another `@Nullable` annotation, and see if that works?

**Comment by elucash** [comment]
maybe the problem is that annotation have `TYPE_USE` target, and there are many problems (at least for us, at least in past) of using those in annotation detection routines. For collections, I would suggest using custom annotations (BYOA - bring your own annotations) with simple names `AllowNulls` and `SkipNulls` which can be placed on the collection. (maybe those can work on the type too). Both Javac and ECJ are (or were) riddled with different annotation processing bugs where compilers just don't provide accurate information about type annotations depending on where and how they are used. It worked on some level, if not anymore – I don't have enough time to troubleshoot it now.
You can try to dig why existing logic doesn't work by starting at `ValueAttribute` class, look for fields `nullElements`, `nullability`, how they are assigned, and then `TypeStringProvider` class which employs all thinkable and unthinkable tricks to get trough partially compiled type descriptors and their annotations. Maybe something didn't work in the first place, I dunno. You can use `protoclass().environment().processing().getMessager().printMessage(Diagnostic.Kind.MANDATORY_WARNING, "DEBUG")` as a way to do debug printing from there.

**Comment by marc-guenther** [comment]
Yes, it seems `TYPE_USE` is the problem. I defined my own Annotation, but as I need it in this exact location, I need to decare it as `TYPE_USE` and then it exhibits the same problem.
I guess for now I will just do a stupid workaround by defining some specific String instance as a placeholder for `null`... :(

I tried to look at the source code, but current `master` doesn't compile:
```
$ mvn clean package
.....
[INFO] Running org.immutables.generator.fixture.WhitespacerTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.012 s <<< FAILURE! - in org.immutables.generator.fixture.WhitespacerTest
[ERROR] run  Time elapsed: 0.007 s  <<< ERROR!
java.lang.Error: 
Unresolved compilation problem: 
	Generator_Whitespacer cannot be resolved to a type
	at org.immutables.generator.fixture.WhitespacerTest.run(WhitespacerTest.java:23)
```

Opening it in Eclipse results in tons of "... cannot be resolved to a type" errors.

I tried the version I was using, but there is some confusion about what is the latest version. In Git, it is `2.8.8`, in documentation is says `2.8.2` and in the Changelog, it is `2.8.1`.

I tried Version `2.8.8` and it cannot download an artifact:
```[ERROR] Failed to execute goal on project value-fixture: Could not resolve dependencies for project org.immutables:value-fixture:jar:2.8.8: Failed to collect dependencies at com.atlassian.fugue:fugue:jar:2.6.1: Failed to read artifact descriptor for com.atlassian.fugue:fugue:jar:2.6.1: Could not transfer artifact com.atlassian.fugue:fugue:pom:2.6.1 from/to spring-milestones (https://repo.spring.io/libs-milestone/): Authentication failed for https://repo.spring.io/libs-milestone/com/atlassian/fugue/fugue/2.6.1/fugue-2.6.1.pom 401 Unauthorized -> [Help 1]
```
 Same for 2.8.2, and 2.8.1. `mvn eclipse:eclipse` has the same problem.

I also upgrade my example to 2.8.8, but it shows the same problem...

**Comment by elucash** [comment]
https://github.com/immutables/immutables/runs/2153947133 shows that last commit to master was passing an "all" JDKs.
try `mvn clean install`, IIRC `package` had problems of using annotation processors compiled in other modules
